examples/demo.rs

```diff
@@ -0,0 +1,233 @@
+        peripherals.pins.gpio10,
+    )?;
+
+    let start_position = DoorPosition::Neutral;
+    let servo_delay = 12;
+
+    let mut servo = ServoHandler::new(servo_driver, start_position, servo_delay);
+    FreeRtos::delay_ms(100);
+
+    let mut client = reqwesp::Client::new()?;
+    // Endpoint for testing REST requests
+    let url = "https://app.lynx-locks.com/api/doors/unlocked/1";
+
+    log::info!("Waiting for authorized credentials...");
+    loop {
+        let mut req = client.get(url);
+        let res = req.send()?;
+
+        if let StatusCode::OK = res.status() {
+            log::info!("Door unlocked!");
+            unlock(&mut led, &mut servo)?;
+        }
+
+        match ykhmac::wait_for_yubikey(Duration::from_millis(1000)) {
+            YubiKeyResult::IsYubiKey => {
+                log::info!("YubiKey detected!");
+                log::info!("Firmware version: {}", ykhmac::get_version());
+                let serial = ykhmac::get_serial();
+                log::info!("Serial number: {serial}");
+                match ykhmac::authenticate() {
+                    AuthStatus::AccessGranted => {
+                        let url =
+                            format!("https://app.lynx-locks.com/api/auth/authorize/1/{serial}");
+                        let mut req = client.get(url.as_str());
+                        let res = req.send()?;
+
+                        if let StatusCode::OK = res.status() {
+                            log::info!("Door unlocked!");
+                            unlock(&mut led, &mut servo)?;
+                        } else {
+                            log::info!("Access Denied");
+                            set_red(&mut led, 3000)?
+                        }
+                    }
+                    AuthStatus::AccessDenied => set_red(&mut led, 3000)?,
+                    AuthStatus::Error(e) => log::warn!("Auth error: {e:?}"),
+                }
+            }
+            YubiKeyResult::NotYubiKey => set_red(&mut led, 3000)?, // Set LED to red for 3 seconds.
+            YubiKeyResult::Error(_) => {}
+        }
+    }
+}
+
+fn unlock(led: &mut Led, servo: &mut ServoHandler) -> std::result::Result<(), LedError> {
+    led.set_color(0x00, 0x10, 0x00)?;
+    servo.set_position(DoorPosition::Unlocked);
+    FreeRtos::delay_ms(5000);
+    led.set_color(0x00, 0x00, 0x00)?;
+    servo.set_position(DoorPosition::Locked);
+    servo.set_position(DoorPosition::Neutral);
+    Ok(())
+}
+
+fn set_red(led: &mut Led, wait_ms: u32) -> std::result::Result<(), LedError> {
+    // Set LED to green for 3 seconds, then off.
+    led.set_color(0x10, 0x00, 0x00)?;
+    FreeRtos::delay_ms(wait_ms);
+    led.set_color(0x00, 0x00, 0x00)
+}
+
+#[derive(Clone, Copy, Debug)]
+enum DoorPosition {
+    Neutral = 90,
+    Unlocked = 37,
+    Locked = 135,
+}
+
+struct ServoHandler<'a> {
+    servo: LedcDriver<'a>,
+    current_position: u32,
+    max_duty: u32,
+    servo_delay: u32,
+}
+
+impl<'a> ServoHandler<'a> {
```

Comment:
im confused are we redefining servo in demo.rs?
should we not just have this on a separate file if its gonna stay consistent?

Reply:
Most of the servo functionality will be exported to a separate module later. Right now, it relies on hard-coded values for the door angles, but I would like the module to be more agnostic to our specific usage (as all the other driver modules are).
This PR is already quite big, so I'd prefer to put that change in a separate one.